src/components/add/Add.js

```diff
@@ -1,24 +1,39 @@
+import { useRef } from 'react';
+import { useDispatch } from 'react-redux';
+import { addBook } from '../../redux/books/booksArrSlice';
 import Button from '../utils/Buttons';
 import Input from '../utils/Input';
 
-const AddNewBook = () => (
-  <>
+const AddNewBook = () => {
+  const dispatch = useDispatch();
+  const titleRef = useRef(null);
+  const authorRef = useRef(null);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const title = titleRef.current?.value;
+    const author = authorRef.current?.value;
```

Comment:
- [ ] Kindly consider adding the local state for title and author, which is updated through the onChange event to fullfill the below requirements 
> Add a <Button> component, which includes:
- > An event handler that adds a book to the books array (with attributes id, title and author)




hint 👇 

```
import React from "react"
import { useDispatch } from "react-redux"
import { addBook } from "path/to/booksSlice"

function NewBookForm() {
  const [title, setTitle] = useState("")
  const [author, setAuthor] = useState("")
  const books = useSelector((state) => state.books.books)
  const dispatch = useDispatch()

  // Other logic that controls form state

  return (
    <div>
      {/* Rest of form JSX */}
      <Button onClick={() => dispatch(addBook({ title, author }))}>
        Add book
      </Button>
    </div>
  )
}

export default NewBookForm

```

while doing so consider fixing this issue when I add a book the title and author are not being fixed check the image below for more clarification
<img width="1361" alt="Screen Shot 2023-06-16 at 18 26 14" src="https://github.com/waelelsafty07/Book-Store/assets/60146030/37e03fc8-5517-45de-996f-4596d2e8817d">

Reply:
@divinecharlotte  thank you